src/OSPSuite.Core/Services/PredictedVsObservedChartService.cs

```diff
@@ -24,20 +23,21 @@ public interface IPredictedVsObservedChartService
       ///    Adds the identity curve to the <paramref name="chart" /> where the values predicted equals the values observed
       ///    <returns>Any identity repositories created</returns>
       /// </summary>
-      IReadOnlyList<DataRepository> AddIdentityCurves(IEnumerable<DataColumn> observationColumns, ParameterIdentificationPredictedVsObservedChart chart);
+      IReadOnlyList<DataRepository> AddIdentityCurves(IEnumerable<DataColumn> observationColumns, PredictedVsObservedChart chart);
 
       /// <summary>
       ///    Adds curves to the <paramref name="chart" /> for each combination of <paramref name="observationColumns" /> and
       ///    <paramref name="calculationColumn" />
       ///    <paramref name="action" /> will be run when the curve is created for the <paramref name="calculationColumn" />
       /// </summary>
-      void AddCurvesFor(IEnumerable<DataColumn> observationColumns, DataColumn calculationColumn, ParameterIdentificationPredictedVsObservedChart chart, Action<DataColumn, Curve> action = null);
+      void AddCurvesFor(IEnumerable<DataColumn> observationColumns, DataColumn calculationColumn, PredictedVsObservedChart chart,
+         Action<DataColumn, Curve> action = null);
 
       /// <summary>
       ///    Sets the <paramref name="chart" /> x axis dimension to the most occurring dimension from
-      ///    <paramref name="observationColumns" />
+      ///    <paramref name="observationColumns" /> and adjusts the axes names
       /// </summary>
-      void SetXAxisDimension(IEnumerable<DataColumn> observationColumns, ParameterIdentificationPredictedVsObservedChart chart);
+      void ConfigureAxesDimensionAndTitle(IEnumerable<DataColumn> observationColumns, PredictedVsObservedChart chart);
```

Comment:
Make this methods use IReadOnly instead of IEnumerable. You are always doing the cast in ToList before anyways it seems

Reply:
done